### PR TITLE
fix(swap): ignore gas swaps for single asset ccm swaps [WEB-1424]

### PR DESCRIPTION
**Pull request by GabrielBuragev**

### CODE REVIEW

File: packages/swap/src/routes/v2/swap.ts [hunk 1]
```diff
@@ -151,7 +155,6 @@ router.get(
       getRequiredBlockConfirmations(internalSrcAsset),
     ]);
 
-    const showCcm = Boolean(swapDepositChannel?.ccmGasBudget || swapDepositChannel?.ccmMessage);
```
Comment (GabrielBuragev) [inline packages/swap/src/routes/v2/swap.ts:154]: ccm params were only returned for deposit channel ccm swaps but not for SC ccm

File: packages/swap/src/routes/v2/swap.ts [hunk 2]
```diff
@@ -72,37 +74,39 @@ router.get(
...
-        isDca: Boolean(
-          swapDepositChannel?.chunkIntervalBlocks && swapDepositChannel.chunkIntervalBlocks > 1,
-        ),
-        fees: [] as SwapFee[],
-      },
-    );
+    const swaps = swapRequest?.swaps.filter((swap) => swap.type !== 'GAS');
+
+    const rolledSwaps = swaps?.length
+      ? swaps.reduce(
```
Comment (acdibble) [inline packages/swap/src/routes/v2/swap.ts:80]:
```suggestion
    const rolledSwaps = swaps?.reduce(
```

**Review by jerryafr — APPROVED** [review]

File: packages/swap/src/routes/v2/swap.ts (hunk 2 above)
Comment (GabrielBuragev) [reply packages/swap/src/routes/v2/swap.ts:80]: we don't want to return the 0s from the default value in the status response (check line 203). At least that felt most natural to me.

**Review by acdibble — APPROVED** [review]

File: packages/swap/src/routes/v2/swap.ts (hunk 2 above)
Comment (GabrielBuragev) [reply packages/swap/src/routes/v2/swap.ts:80]: We just had an ingress -> gas swap (for the egress fee) -> egress . No principal or other swaps done.

Comment (acdibble) [reply packages/swap/src/routes/v2/swap.ts:80]: yeah but the optional chain will have the same result as this ternary

Comment (GabrielBuragev) [reply packages/swap/src/routes/v2/swap.ts:80]: hmm, not when the array of swaps is empty though? rolledSwaps will be initialized with the default value of the reducer

**Pull request merged**